Lib/test/test_listcomps.py

```diff
@@ -236,6 +241,29 @@ def test_introspecting_frame_locals(self):
         outputs = {"i": 20}
         self._check_in_scopes(code, outputs)
 
+    def test_nested(self):
+        code = """
+            l = [1, 2, 3]
+            y = [x*2 for x in [x for x in l]]
```

Comment:
I'd say this isn't really nested because the inner comprehension is executed in the enclosing scope.

Something like this is more interesting:

`[[x ** 2 for x in range(x)] for x in l]`

`[x for [x ** x for x in range(x)][x] for x in l]`

Reply:
Good point :) Pushed your first example as a better test.

The second example is invalid syntax as written; if there's a distinct case you were aiming to get at there, let me know and I can try to craft a test for it.

Reply:
Sorry, I wrote that without testing.

Here's a version of the second example that runs:

```
>>> l = [1, 2, 3]
>>> x = 3
>>> [x for [x ** x for x in range(x)][x - 1] in l]
[3, 3, 3]
```

The point is to test cases where the nested comprehension is in the assignment target for the outer comprehension.

Reply:
Even worse:

```
>>> l = [(1, 2), (3, 4), (5, 6)]
>>> [x for (x, [x ** x for x in range(x)][x - 1]) in l]
[1, 3, 5]
```
(Now both comprehensions assign to the name `x`.)

Reply:
Thanks for the great examples! That latter example looks like it uncovered a real incompatibility, in class/module scope only. Looking into it now.

Reply:
Now I need to figure out how to work one of these terrible snippets into a PR to one of Jelle's projects :)